### PR TITLE
add recycle for unbinding event

### PR DESCRIPTION
**Pull request by Jchou24**

### CODE REVIEW

File: js/reveal.js [hunk 1]
```diff
@@ -1267,19 +1281,7 @@
...
-				// Make sure we're dealing with JSON
-				if( typeof data === 'string' && data.charAt( 0 ) === '{' && data.charAt( data.length - 1 ) === '}' ) {
-					data = JSON.parse( data );
-
-					// Check if the requested method can be found
-					if( data.method && typeof Reveal[data.method] === 'function' ) {
-						Reveal[data.method].apply( Reveal, data.args );
-					}
-				}
-			}, false );
```
Comment (Itchibon777) [inline js/reveal.js]: window.addEventListener( 'message', function ( event ) {
				var data = event.data;

				// Make sure we're dealing with JSON
				if( typeof data === 'string' && data.charAt( 0 ) === '{' && data.charAt( data.length - 1 ) === '}' ) {
					data = JSON.parse( data );

					// Check if the requested method can be found
					if( data.method && typeof Reveal[data.method] === 'function' ) {
						Reveal[data.method].apply( Reveal, data.args );
					}
				}
			}, false );

**Comment by hakimel** [comment]
Hey, we already have a `Reveal.removeEventListeners()` for this so there shouldn't be any need to add a new recycle method: https://github.com/hakimel/reveal.js/blob/master/js/reveal.js#L1559-L1595

**Comment by Jchou24** [comment]
Hi, thanks for your reply

Sorry, my point is I do not find the events in recycle() to be removed in Reveal.removeEventListeners()
the events are:
	window.removeEventListener( 'load', layout, false )
	window.removeEventListener( 'load', setupPDF );
	window.removeEventListener( 'message', HandleMessage, false );
	window.removeEventListener( 'hashchange', onWindowHashChange, false );
	window.removeEventListener( 'resize', onWindowResize, false );
	window.removeEventListener( 'load', removeAddressBar, false );
	window.removeEventListener( 'orientationchange', removeAddressBar, false );

and I am not sure why the events not removed in Reveal.removeEventListeners()
so I create recycle() to avoid influencing Reveal.removeEventListeners()

**Comment by hakimel** [comment]
I see. I'll review the list of events in removeEventListeners to make sure everything necessary is covered. Thanks!

**Pull request closed without merging**